interactors/singleTransactionAddressNonceHandler_test.go

```diff
@@ -0,0 +1,238 @@
+		addressHandler := data.NewAddressFromBytes(pubkey)
+
+		_, err := NewSingleTransactionAddressNonceHandler(&testsCommon.ProxyStub{}, addressHandler)
+		assert.Nil(t, err)
+	})
+}
+
+func TestSingleTransactionAddressNonceHandler_ApplyNonce(t *testing.T) {
+	t.Parallel()
+
+	t.Run("proxy returns error should error", func(t *testing.T) {
+
+		proxy := &testsCommon.ProxyStub{
+			GetAccountCalled: func(address core.AddressHandler) (*data.Account, error) {
+				return nil, expectedErr
+			},
+		}
+
+		anh, _ := NewSingleTransactionAddressNonceHandler(proxy, testAddress)
+
+		txArgs := createTxArgs()
+
+		err := anh.ApplyNonce(context.Background(), &txArgs)
+		require.Equal(t, expectedErr, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+
+		blockchainNonce := uint64(100)
+		proxy := &testsCommon.ProxyStub{
+			GetAccountCalled: func(address core.AddressHandler) (*data.Account, error) {
+				return &data.Account{Nonce: blockchainNonce}, nil
+			},
+		}
+
+		anh, _ := NewSingleTransactionAddressNonceHandler(proxy, testAddress)
+
+		txArgs := createTxArgs()
+
+		err := anh.ApplyNonce(context.Background(), &txArgs)
+		require.Nil(t, err)
+		require.Equal(t, blockchainNonce, txArgs.Nonce)
+	})
+}
+
+func TestSingleTransactionAddressNonceHandler_fetchGasPriceIfRequired(t *testing.T) {
+	t.Parallel()
+
+	t.Run("proxy returns error should set invalid gasPrice(0)", func(t *testing.T) {
+		proxy := &testsCommon.ProxyStub{
+			GetNetworkConfigCalled: func() (*data.NetworkConfig, error) {
+				return nil, expectedErr
+			},
+		}
+		anh, _ := NewSingleTransactionAddressNonceHandler(proxy, testAddress)
+		anh.gasPrice = 100000
+		anh.nonceUntilGasIncreased = 100
+
+		anh.fetchGasPriceIfRequired(context.Background(), 101)
+		require.Equal(t, uint64(0), anh.gasPrice)
+	})
+}
+
+func TestSingleTransactionAddressNonceHandler_DropTransactions(t *testing.T) {
+	t.Parallel()
+
+	txArgs := createTxArgs()
+
+	blockchainNonce := uint64(100)
+	minGasPrice := uint64(10)
+	proxy := &testsCommon.ProxyStub{
+		GetAccountCalled: func(address core.AddressHandler) (*data.Account, error) {
+			return &data.Account{Nonce: blockchainNonce}, nil
+		},
+		GetNetworkConfigCalled: func() (*data.NetworkConfig, error) {
+			return &data.NetworkConfig{MinGasPrice: minGasPrice}, nil
+		},
+	}
+
+	anh, _ := NewSingleTransactionAddressNonceHandler(proxy, testAddress)
+
+	err := anh.ApplyNonce(context.Background(), &txArgs)
+	require.Nil(t, err)
+
+	tx := createTx(txArgs.GasPrice, txArgs)
+	_, err = anh.SendTransaction(context.Background(), tx)
+	require.Nil(t, err)
+
+	require.Equal(t, uint64(0), anh.nonceUntilGasIncreased)
+	require.Equal(t, minGasPrice, anh.gasPrice)
+	require.NotNil(t, anh.transaction)
+
+	anh.DropTransactions()
+
+	require.Equal(t, blockchainNonce, anh.nonceUntilGasIncreased)
+	require.Equal(t, minGasPrice+1, anh.gasPrice)
+	require.Nil(t, anh.transaction)
+}
+
+func TestSingleTransactionAddressNonceHandler_SendTransaction(t *testing.T) {
+	t.Parallel()
```

Comment:
can be moved inside t.Run

Reply:
as discussed, we`ll make the t.Run just a simple comment